api/src/main/java/org/openmrs/api/impl/AdministrationServiceImpl.java

```diff
@@ -144,34 +147,62 @@ public SortedMap<String, String> getSystemVariables() throws APIException {
 	/**
 	 * @see org.openmrs.api.AdministrationService#getGlobalProperty(java.lang.String)
 	 */
+	
 	@Override
 	@Transactional(readOnly = true)
 	public String getGlobalProperty(String propertyName) throws APIException {
-		// This method should not have any authorization check
 		if (propertyName == null) {
 			return null;
 		}
 		
+		User user = Context.getAuthenticatedUser();
 		GlobalProperty gp = dao.getGlobalPropertyObject(propertyName);
+		
+		try {
+			Context.addProxyPrivilege(PrivilegeConstants.GET_GLOBAL_PROPERTIES);
+			if (user == null && !isAnonymouslyAccessible(gp)) {
+				log.warn("Property '{}' is not accessible anonymously.", propertyName);
+				throw new APIAuthenticationException("GlobalProperty.property.notAnonymous");
+			}
+		}
+		finally {
+			Context.removeProxyPrivilege(PrivilegeConstants.GET_GLOBAL_PROPERTIES);
+		}
+		
+
 		if (gp != null) {
 			if (canViewGlobalProperty(gp)) {
 				return gp.getPropertyValue();
 			} else {
-				throw new APIException("GlobalProperty.error.privilege.required.view", new Object[] {
-					gp.getViewPrivilege().getPrivilege(), propertyName });
+				throw new APIException("GlobalProperty.error.privilege.required.view",
+				        new Object[] { gp.getViewPrivilege().getPrivilege(), propertyName });
 			}
 		} else {
 			return null;
 		}
+		
+	}
+	
+	String[] anonymousArray = { "owa.appBaseUrl", "login.url", "spa.baseUrl", "timezone.conversions", "default_theme",
```

Comment:
All of these should be moved to the top of the file with the remainder of the properties.